src/api/clients/BaseApiClient.ts

```diff
@@ -0,0 +1,99 @@
+import axios, { AxiosError } from "axios";
+import { ApiError } from "../types/ApiResponses";
+import { HttpMethod } from "../types/Enums";
+
+export abstract class BaseApiClient{
+
+    authToken: string;
+
+    constructor(authToken: string){
+        this.authToken = authToken;
+    }
+
+    /**
+     * Gets a paged response from supporting api endpoints
+     * @param url The api url
+     * @param pageNumber The page number
+     * @param pageSize The page size
+     * @returns A Paged response of type TResponse if successful. An ApiError if not successful.
+     */
+    protected async getPagedResponse<TResponse>(
+        url: string, 
+        pageNumber: number = 1, 
+        pageSize: number = 20
+        ): Promise<{
+            data?: TResponse;
+            error?: ApiError
+        }> 
+    {
+        url = `${url}?page=${pageNumber}&size=${pageSize}`;
+        
+        return await this.httpRequest<TResponse>(url, HttpMethod.GET);
+    };
+
+    /**
+     * Performs a http request to the MoneyMoov api.
+     * @param url The request url
+     * @param method The Http Method.
+     * @param postData Optional. The data to post if specified.
+     * @returns A response of type TResponse if successful. An ApiError if not successful.
+     */
+    protected async httpRequest<TResponse>(
+        url: string,
+        method: HttpMethod,
+        postData?: any,
+        ): Promise<{
+            data?: TResponse;
+            error?: ApiError
+        }> {
+
+            console.log(`Requesting: ${method} ${url}`);
+
+            let contentType = 'application/json';
+
+            // Send form encoding on POST and PUT
+            // Axios will automatically serialize the postData object to form urlencoded format
+            if (method === HttpMethod.POST || method === HttpMethod.PUT){
+                contentType = 'application/x-www-form-urlencoded';
+            }
```

Comment:
Might be easier to use JSON encoding?

Reply:
Currently not all of our controller actions support it. Axios will take the payload and serialize it for a form.